syft/src/main/java/org/openmined/syft/domain/JobRepository.kt

```diff
@@ -19,25 +18,22 @@ internal const val PLAN_OP_TYPE = "torchscript"
 private const val TAG = "JobDownloader"
 
 @ExperimentalUnsignedTypes
-internal class JobRepository(
-    private val jobLocalDataSource: JobLocalDataSource,
-    private val jobRemoteDataSource: JobRemoteDataSource
-) {
+internal class JobRepository(val dataSource: JobDataSource) {
 
     private val trainingParamsStatus = AtomicReference(DownloadStatus.NOT_STARTED)
     val status: DownloadStatus
         get() = trainingParamsStatus.get()
 
     fun getDiffScript(config: SyftConfiguration) =
-            jobLocalDataSource.getDiffScript(config)
+            dataSource.localDataSource.getDiffScript(config)
 
     fun persistToLocalStorage(
         input: InputStream,
         parentDir: String,
         fileName: String,
         overwrite: Boolean = false
     ): String {
-        return jobLocalDataSource.save(input, parentDir, fileName, overwrite)
+        return dataSource.localDataSource.save(input, parentDir, fileName, overwrite)
```

Comment:
Always call repository.save repository will handle how to use datasource. Datasource objects should not be needed outside repository

Reply:
you mean to encapsulate the logic of localDataSource inside the JobDataSource? and serve the JobDataSource as a repository for local data access?

Reply:
Nope I mean remove the jobdatasource. Have localdatasource and remote data source only. Let repository decide which one to call. A jobdatasource is redundant